src/terms/nonlocal.jl

```diff
@@ -108,7 +108,7 @@ function build_projection_coefficients_(T, psps, psp_positions)
     end  # psp, r
     @assert count == n_proj
 
-    proj_coeffs
+    to_device(architecture, proj_coeffs)
```

Comment:
I would move this to the calling side (and have the function always return on the CPU).